### PR TITLE
backup-and-restore: add basic introduction

### PR DESCRIPTION
**Pull request by 3pointer**
### What is changed, added or deleted? 

add basic introduction for br

### What is the related PR or file link(s)? 
N/A

N/A
### Which version does your change affect? 
N/A

### CODE REVIEW

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 1]
```diff
@@ -12,6 +12,49 @@ Backup & Restore（以下简称 BR）是 TiDB 分布式备份恢复的命令行
 
 BR 是分布式备份恢复的工具，它将备份和恢复操作命令下发到各个 TiKV 节点。TiKV 收到命令后执行相应的备份和恢复操作。在一次备份或恢复中，各个 TiKV 节点都会有一个项目的备份路径，TiKV 备份时产生的备份文件将会保存在该路径下，恢复时也会从该路径读取相应的备份文件。
 
+### 备份原理
+
+BR 执行备份时，会先从 PD 获取到
+
+- 当前的 TS 作为备份快照的时间点
+- 当前集群的 TikV 节点信息
```
Comment (overvenus) [inline dev/how-to/maintain/backup-and-restore/br.md]:
```suggestion
- 当前集群的 TiKV 节点信息
```

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 2]
```diff
@@ -12,6 +12,48 @@ Backup & Restore（以下简称 BR）是 TiDB 分布式备份恢复的命令行
...
+
+然后根据上述信息，在 BR 内部启动一个 TiDB，获取对应 TS 的数据库／表信息。同时过滤掉系统库（`information_schema`，`performance_schema`，`mysql`)，
```
Comment (overvenus) [inline dev/how-to/maintain/backup-and-restore/br.md]:
```suggestion
然后根据上述信息，在 BR 内部调用 TiDB 相关接口，获取对应 TS 的数据库／表信息。同时过滤掉系统库（`information_schema`，`performance_schema`，`mysql`)，
```

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 3]
```diff
@@ -12,6 +12,48 @@ Backup & Restore（以下简称 BR）是 TiDB 分布式备份恢复的命令行
...
 BR 是分布式备份恢复的工具，它将备份和恢复操作命令下发到各个 TiKV 节点。TiKV 收到命令后执行相应的备份和恢复操作。在一次备份或恢复中，各个 TiKV 节点都会有一个项目的备份路径，TiKV 备份时产生的备份文件将会保存在该路径下，恢复时也会从该路径读取相应的备份文件。
 
+### 备份原理
+
+BR 执行备份时，会先从 PD 获取到
+
+- 当前的 TS 作为备份快照的时间点
+- 当前集群的 TiKV 节点信息
+
+然后根据上述信息，在 BR 内部调用 TiDB 相关接口，获取对应 TS 的数据库／表信息。同时过滤掉系统库（`information_schema`，`performance_schema`，`mysql`)，
```
Comment (GregoryIan) [inline dev/how-to/maintain/backup-and-restore/br.md]:
```suggestion
然后根据上述信息，在 BR 内部调用 TiDB 相关接口，获取对应 TS snapshot 的数据库／表信息。同时过滤掉系统库（`information_schema`，`performance_schema`，`mysql`)，
```

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 4]
```diff
@@ -12,6 +12,48 @@ Backup & Restore（以下简称 BR）是 TiDB 分布式备份恢复的命令行
...
+    EndKey:       endKey,      // 备份结束点，endKey 不会被备份
+    StartVersion: backupTS,    // 备份快照时间点
+    ...
+    Path:         path,        // 备份文件存储路径
+    RateLimit:    rateLimit,   // 备份速度，单位 MB/s
+    Concurrency:  concurrency, // 执行备份文件的线程数，默认是 4
+}
+```
+
+TiKV 节点收到备份请求后，会遍历自己的全部 Region Leader，找到和请求中 KV Range 有重叠的范围，将该范围下的部分或者全部数据进行备份，这样在备份路径下生成对应的 SST 文件（命名格式是 storeID_regionID_regionEpoch_tableID)。
```
Comment (GregoryIan) [inline dev/how-to/maintain/backup-and-restore/br.md]:
```suggestion
TiKV 节点收到备份请求后，会遍历自己的全部 Region Leader，找到和请求中 KV Range 有重叠的范围的 region，将该范围下的部分或者全部数据进行备份，在备份路径下生成对应的 SST 文件（命名格式是 storeID_regionID_regionEpoch_tableID)。
```

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 5]
```diff
@@ -12,6 +12,48 @@ Backup & Restore（以下简称 BR）是 TiDB 分布式备份恢复的命令行
...
+TiKV 节点在备份完每一个 Region Leader 后将元信息返回给 BR，BR 将这些元信息收集存储进 backupMeta 文件中，等待恢复时使用。
```
Comment (GregoryIan) [inline dev/how-to/maintain/backup-and-restore/br.md]:
```suggestion
TiKV 节点在备份完对应 Region Leader 的数据后将元信息返回给 BR，BR 将这些元信息收集存储进 backupMeta 文件中，等待恢复时使用。
```

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 6]
```diff
@@ -12,6 +12,48 @@ Backup & Restore（以下简称 BR）是 TiDB 分布式备份恢复的命令行
...
+如果执行命令时开启了 checksum，那么 BR 在最后会对备份的每一张表计算 checksum 用于校验。
+
+### 恢复原理
+
+BR 执行恢复时，首先解析备份路径下的 backupMeta 文件
+
+- 根据解析出来的库表信息，启动一个 TiDB 实例在新集群创建对应的库表
```
Comment (GregoryIan) [inline dev/how-to/maintain/backup-and-restore/br.md]:
```suggestion
- 根据解析出来的库表信息，在自己内部启动一个 TiDB 实例在新集群创建对应的库表
```

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 7]
```diff
@@ -12,6 +12,57 @@ Backup & Restore（以下简称 BR）是 TiDB 分布式备份恢复的命令行
 
 BR 是分布式备份恢复的工具，它将备份和恢复操作命令下发到各个 TiKV 节点。TiKV 收到命令后执行相应的备份和恢复操作。在一次备份或恢复中，各个 TiKV 节点都会有一个项目的备份路径，TiKV 备份时产生的备份文件将会保存在该路径下，恢复时也会从该路径读取相应的备份文件。
 
+### 备份原理
+
+BR 执行备份时，会先从 PD 获取到
```
Comment (leoppro) [inline dev/how-to/maintain/backup-and-restore/br.md]:
```suggestion
BR 执行备份时，会先从 PD 获取到：
```

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 8]
```diff
@@ -12,6 +12,57 @@ Backup & Restore（以下简称 BR）是 TiDB 分布式备份恢复的命令行
...
+### 备份原理
+
+BR 执行备份时，会先从 PD 获取到：
+
+- 当前的 TS 作为备份快照的时间点。
+- 当前集群的 TikV 节点信息。
+
+然后根据上述信息，在 BR 内部启动一个 TiDB，获取对应 TS 的数据库／表信息。同时过滤掉系统库（`information_schema`，`performance_schema`，`mysql`)；
+
+如果是全量备份，会遍历全部库表，并且根据每一张表构建需要备份的 KV Range；
```
Comment (leoppro) [inline dev/how-to/maintain/backup-and-restore/br.md]:
```suggestion
- 如果是全量备份，会遍历全部库表，并且根据每一张表构建需要备份的 KV Range；
```

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 9]
```diff
@@ -12,6 +12,57 @@ Backup & Restore（以下简称 BR）是 TiDB 分布式备份恢复的命令行
...
+
+如果是单表备份，会根据该表构建需要备份的 KV Range。
```
Comment (leoppro) [inline dev/how-to/maintain/backup-and-restore/br.md]:
```suggestion
- 如果是单表备份，会根据该表构建需要备份的 KV Range。
```

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 10]
```diff
@@ -12,6 +12,57 @@ Backup & Restore（以下简称 BR）是 TiDB 分布式备份恢复的命令行
...
 BR 是分布式备份恢复的工具，它将备份和恢复操作命令下发到各个 TiKV 节点。TiKV 收到命令后执行相应的备份和恢复操作。在一次备份或恢复中，各个 TiKV 节点都会有一个项目的备份路径，TiKV 备份时产生的备份文件将会保存在该路径下，恢复时也会从该路径读取相应的备份文件。
 
+### 备份原理
+
+BR 执行备份时，会先从 PD 获取到：
+
+- 当前的 TS 作为备份快照的时间点。
+- 当前集群的 TikV 节点信息。
+
+然后根据上述信息，在 BR 内部启动一个 TiDB，获取对应 TS 的数据库／表信息。同时过滤掉系统库（`information_schema`，`performance_schema`，`mysql`)；
```
Comment (leoppro) [inline dev/how-to/maintain/backup-and-restore/br.md]:
```suggestion
然后根据上述信息，在 BR 内部启动一个 TiDB，获取对应 TS 的数据库／表信息。同时过滤掉系统库（`information_schema`，`performance_schema`，`mysql`)。
```

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 11]
```diff
@@ -12,6 +12,57 @@ Backup & Restore（以下简称 BR）是 TiDB 分布式备份恢复的命令行
...
+
+如果是全量备份，会遍历全部库表，并且根据每一张表构建需要备份的 KV Range；
+
+如果是单表备份，会根据该表构建需要备份的 KV Range。
+
+最后，BR 将需要备份的 KV Range 收集后，构造完整请求分发给集群内的 TiKV 节点。
+
+请求主要结构如下
```
Comment (leoppro) [inline dev/how-to/maintain/backup-and-restore/br.md]:
```suggestion
请求主要结构如下：
```

**Comment by lilin90** [comment]
@TomShawn Please take a look at the language and apply changes to v3.1. Thanks!
@3pointer Any further changes to this BR doc, please let @TomShawn know because he is working on translating it into English. And please also update `v3.1`. Thanks!

**Pull request merged**